cmd/clusterctl/client/repository/template_client_test.go

```diff
@@ -184,3 +218,40 @@ func Test_templates_Get(t *testing.T) {
 		})
 	}
 }
+
+type FakeProcessor struct {
```

Comment:
Hahaha...yeah I tried really hard doing this  many times. 😆 

I would run into cyclic import issues because the `YamlProcessor` interface has `Process([]byte, config.VariablesClient) ([]byte, error)`. And the `config` package imports `internal/test` (since we don't use separate `_test` packages).

Even if I created my own interface for the variables client, it would have to be in a package that doesn't import `internal/test` which seemed wrong.

I was very sad 😞 to have to re-implement these fake processors in the different places.

Reply:
Looks like having a dependency on config for re-using an interface that exposes a single func is a bad idea...
What about having a YamlProcessor func interface, and somehere making an asserting the two interfaces actually met?

Reply:
> Looks like having a dependency on config for re-using an interface that exposes a single func is a bad idea...

I wish I thought of this sooner. 😄 

Yeah...so I was thinking about something like 
`Process([]byte, func(string) string) ([]byte, error)`. That's the most generic form of a "Getter" I could think of.

This way we could implement whatever getter we want.
```
varsGetter := func(key string) string {
  value, _ := configVariablesClient.Get(key)
  return value
}
processedYaml, err := Process(rawArtifact, varsGetter)
```

WDYT about this style?